### PR TITLE
Remove support of v1 proofs

### PR DESCRIPTION
**Pull request by elmattic**
This is the counterpart of [PR 1391](https://github.com/filecoin-project/builtin-actors/pull/1391) from builtin actors repository.

### CODE REVIEW

**Review by arajasek — APPROVED** [review]
LGTM

**Comment by Stebalien** [comment]
@LesnyRumcajs before I cut a release, I realize this might cause issues in Forest. Is this change fine?

File: shared/src/sector/registered_proof.rs [hunk 1]
```diff
@@ -115,16 +115,6 @@ impl Default for RegisteredSealProof {
 #[derive(PartialEq, Eq, Copy, Clone, Debug, Hash)]
 #[cfg_attr(feature = "arb", derive(arbitrary::Arbitrary))]
 pub enum RegisteredPoStProof {
-    StackedDRGWinning2KiBV1,
-    StackedDRGWinning8MiBV1,
-    StackedDRGWinning512MiBV1,
-    StackedDRGWinning32GiBV1,
-    StackedDRGWinning64GiBV1,
```
Comment (Stebalien) [inline shared/src/sector/registered_proof.rs:122]: Did we mean to remove winning post here?

File: shared/src/sector/registered_proof.rs [hunk 2]
```diff
@@ -253,19 +217,19 @@ impl RegisteredSealProof {
         match self {
             Self::StackedDRG64GiBV1
             | Self::StackedDRG64GiBV1P1
-            | Self::StackedDRG64GiBV1P1_Feat_SyntheticPoRep => Ok(StackedDRGWinning64GiBV1),
+            | Self::StackedDRG64GiBV1P1_Feat_SyntheticPoRep => Ok(StackedDRGWindow64GiBV1P1),
```
Comment (Stebalien) [inline shared/src/sector/registered_proof.rs:220]: Winning v. Window post?

File: shared/src/sector/registered_proof.rs [hunk 3]
```diff
@@ -369,21 +333,11 @@ macro_rules! i64_conversion {
...
-    StackedDRGWindow2KiBV1P1 => 10,
-    StackedDRGWindow8MiBV1P1 => 11,
-    StackedDRGWindow512MiBV1P1 => 12,
-    StackedDRGWindow32GiBV1P1 => 13,
-    StackedDRGWindow64GiBV1P1 => 14,
+    StackedDRGWindow2KiBV1P1 => 0,
+    StackedDRGWindow8MiBV1P1 => 1,
+    StackedDRGWindow512MiBV1P1 => 2,
+    StackedDRGWindow32GiBV1P1 => 3,
+    StackedDRGWindow64GiBV1P1 => 4,
```
Comment (Stebalien) [inline shared/src/sector/registered_proof.rs:340]: We should not be re-assigning numbers.

File: shared/src/sector/registered_proof.rs (hunk 1 above)
Comment (elmattic) [reply shared/src/sector/registered_proof.rs:122]: I don't think so.

File: shared/src/sector/registered_proof.rs (hunk 3 above)
Comment (elmattic) [reply shared/src/sector/registered_proof.rs:340]: Agree. This was more or less done so we can drop array size in builtin-actors repository.(`REGISTERED_POST_PROOF_VARIANTS`).

File: shared/src/sector/registered_proof.rs (hunk 2 above)
Comment (elmattic) [reply shared/src/sector/registered_proof.rs:220]: Should be Winning according to the doc comment.

**Comment by elmattic** [comment]
@Stebalien Let me open a new PR to address your comments, ok?

File: shared/src/sector/registered_proof.rs (hunk 3 above)
Comment (arajasek) [reply shared/src/sector/registered_proof.rs:340]: Yeah, unfortunately we can't do that (see [here](https://github.com/filecoin-project/builtin-actors/pull/1391#discussion_r1340149993)).

File: shared/src/sector/registered_proof.rs (hunk 2 above)
Comment (arajasek) [reply shared/src/sector/registered_proof.rs:220]: Can we consider deleting this method entirely? I'm guessing it's used in Forest, but can we make it a helper method there? The FVM really doesn't care about winning posts.

**Review by arajasek — COMMENTED** [review]
Sorry, dropped the ball on review here. Thanks for fixing up @elmattic !

File: shared/src/sector/registered_proof.rs (hunk 2 above)
Comment (elmattic) [reply shared/src/sector/registered_proof.rs:220]: I have nothing against. Let's remove it.

**Pull request merged**